File: DemoMethods/Advanced/MoreLikeThis.cs
using System;
using System.Linq;
using System.Web.Http;
using DemoMethods.Entities;
using DemoMethods.Helpers;
using Raven.Abstractions.Data;
using Raven.Client.Bundles.MoreLikeThis;

namespace DemoMethods.Advanced
{
    public partial class AdvancedController : DemoApiController
    {
        // [HttpGet]
        public object MoreLikeThis(string documentId = "lastfm/9295")
        {
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                LastFm[] mltByArtist = session
                    .Advanced
                    .MoreLikeThis<LastFm>(
                        "IndexFullTextSearch",
                        null,
                        new MoreLikeThisQuery
                        {
                            IndexName = "IndexFullTextSearch",
                            DocumentId = documentId,
                            Fields = new[] { "Artist" }
                        });

                return (mltByArtist.ToList());
            }
        }
    }
}

File: DemoMethods/src/main/java/net/ravendb/demo/advanced/MoreLikeThis.java
package net.ravendb.demo.advanced;

import net.ravendb.abstractions.data.MoreLikeThisQuery;
import net.ravendb.client.IDocumentSession;
import net.ravendb.demo.DocumentStoreHolder;
import net.ravendb.demo.entities.LastFm;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class MoreLikeThis {

    @RequestMapping("/Advanced/MoreLikeThis")
    public LastFm[] moreLikeThis(
            @RequestParam(value = "documentId", defaultValue = "lastfm/9295") String documentId) {

        try (IDocumentSession session = DocumentStoreHolder.getStore().openSession()) {

            MoreLikeThisQuery query = new MoreLikeThisQuery();
            query.setIndexName("IndexFullTextSearch");
            query.setDocumentId(documentId);
            query.setFields(new String[] { "Artist" });

            LastFm[] mltByArtist = session
                    .advanced()
                    .moreLikeThis(LastFm.class, "IndexFullTestSearch", null, query);

            return mltByArtist;
        }
    }
}
